From 27aae4ab444d43c2703ebac3decabc99be5a9704 Mon Sep 17 00:00:00 2001
From: Nitin Garg <nitin.garg@freescale.com>
Date: Fri, 8 Feb 2013 14:44:33 -0600
Subject: [PATCH 161/304] ENGR00244231: Add support Atheros 3k bluetooth for
 JB4.2

Enable support of Atheros 3k bluetooth for JB4.2 bluedroid stack.

Signed-off-by: Nitin Garg <nitin.garg@freescale.com>
---
 imx53_smd/BoardConfig.mk              |  3 ++-
 imx53_smd/bluetooth/bdroid_buildcfg.h | 33 +++++++++++++++++++++++++++++++++
 imx5x/imx5x.mk                        |  3 +++
 3 files changed, 38 insertions(+), 1 deletion(-)
 create mode 100644 imx53_smd/bluetooth/bdroid_buildcfg.h

diff --git a/imx53_smd/BoardConfig.mk b/imx53_smd/BoardConfig.mk
index 244cc53..3c3e15a 100755
--- a/imx53_smd/BoardConfig.mk
+++ b/imx53_smd/BoardConfig.mk
@@ -65,7 +65,8 @@ TARGET_USERIMAGES_USE_EXT4 := true
 
 # atheros 3k BT
 BOARD_USE_AR3K_BLUETOOTH := true
-BOARD_BLUETOOTH_BDROID_BUILDCFG_INCLUDE_DIR := device/generic/common/bluetooth
+BOARD_HAVE_BLUETOOTH_QCOM_ATH3K := true
+BOARD_BLUETOOTH_BDROID_BUILDCFG_INCLUDE_DIR := device/fsl/imx53_smd/bluetooth
 
 USE_OPENGL_RENDERER := true
 BOARD_EGL_CFG := device/fsl-proprietary/gpu/egl/egl.cfg
diff --git a/imx53_smd/bluetooth/bdroid_buildcfg.h b/imx53_smd/bluetooth/bdroid_buildcfg.h
new file mode 100644
index 0000000..7e9890e
--- /dev/null
+++ b/imx53_smd/bluetooth/bdroid_buildcfg.h
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 2012 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* Copyright (C) 2013 Freescale Semiconductor, Inc. */
+
+#ifndef _BDROID_BUILDCFG_H
+#define _BDROID_BUILDCFG_H
+
+#define BTM_DEF_LOCAL_NAME "iMX53 SMD"
+
+// Networking, Capturing, Object Transfer
+// MAJOR CLASS: COMPUTER
+// MINOR CLASS: PALM SIZE PC/PDA
+#define BTA_DM_COD {0x1A, 0x01, 0x14}
+
+#define BTIF_HF_SERVICES (BTA_HSP_SERVICE_MASK)
+#define BTIF_HF_SERVICE_NAMES  { BTIF_HSAG_SERVICE_NAME }
+#define PAN_NAP_DISABLED TRUE
+
+#endif
diff --git a/imx5x/imx5x.mk b/imx5x/imx5x.mk
index 447a2cf..e8f5fd5 100644
--- a/imx5x/imx5x.mk
+++ b/imx5x/imx5x.mk
@@ -15,6 +15,7 @@ PRODUCT_PACKAGES += \
 	PhaseBeam				\
 	HoloSpiralWallpaper			\
 	Gallery2				\
+	LegacyCamera				\
 	SoundRecorder				\
 	FSLOta					\
 	VideoEditor				\
@@ -73,6 +74,8 @@ PRODUCT_PACKAGES += \
 	hwcomposer.imx5x            		\
 	camera.imx5x            		\
 	power.imx5x				\
+	libbt-vendor				\
+	bt_vendor.conf				\
 	magd
 
 # Bluetooth firmware files.
-- 
1.8.0

